axelor-production/src/main/java/com/axelor/apps/production/service/manuforder/ManufOrderServiceImpl.java

```diff
@@ -1323,4 +1324,48 @@ public void checkApplicableManufOrder(ManufOrder manufOrder) throws AxelorExcept
           I18n.get(ProductionExceptionMessage.CHECK_BOM_AND_PROD_PROCESS));
     }
   }
+
+  @Override
+  public Map<Product, BigDecimal> getMissingComponents(ManufOrder manufOrder)
+      throws AxelorException {
+    Map<Product, BigDecimal> missingProductsMap = new HashMap<>();
+    Company company = manufOrder.getCompany();
+    BillOfMaterial billOfMaterial = manufOrder.getBillOfMaterial();
+
+    if (company == null
+        || billOfMaterial == null
+        || billOfMaterial.getQty().compareTo(BigDecimal.ZERO) <= 0
+        || CollectionUtils.isEmpty(billOfMaterial.getBillOfMaterialLineList())) {
+      return missingProductsMap;
+    }
+
+    BigDecimal bomQty = billOfMaterial.getQty();
+    BigDecimal qty = manufOrder.getQty();
+
+    Map<Product, BigDecimal> bomLineMap =
+        billOfMaterial.getBillOfMaterialLineList().stream()
+            .collect(
+                Collectors.groupingBy(
+                    BillOfMaterialLine::getProduct,
+                    Collectors.reducing(
+                        BigDecimal.ZERO, BillOfMaterialLine::getQty, BigDecimal::add)));
+
+    for (Entry<Product, BigDecimal> billOfMaterialLine : bomLineMap.entrySet()) {
+      Product product = billOfMaterialLine.getKey();
+      BigDecimal bomLineQty = billOfMaterialLine.getValue();
+      BigDecimal availableQty = productStockLocationService.getAvailableQty(product, company, null);
+      BigDecimal qtyNeeded = qty.multiply(bomLineQty).divide(bomQty);
```

Comment:
I think we have to have a rounding for the `divide` call. It should work if we set the scale to appBaseService.getNbDecimalDigitForQty(), since it is used after in a subtraction.